src/main/java/com/impactupgrade/nucleus/client/MailchimpClient.java

```diff
@@ -71,10 +72,15 @@ public class MailchimpClient {
   protected static final Integer BATCH_STATUS_MAX_RETRIES = 24;
 
   protected final com.ecwid.maleorang.MailchimpClient client;
+  protected final String apiKey;
   protected final Environment env;
 
   public MailchimpClient(EnvironmentConfig.CommunicationPlatform mailchimpConfig, Environment env) {
-    client = new com.ecwid.maleorang.MailchimpClient(mailchimpConfig.secretKey);
+    String profile = System.getenv("PROFILE");
+    System.out.println("profile: " + profile);
+    apiKey = !Strings.isNullOrEmpty(profile) && profile.toLowerCase(Locale.ROOT).contains("it") ?
+            env.decodeBase64(mailchimpConfig.secretKey) : mailchimpConfig.secretKey;
+    client = new com.ecwid.maleorang.MailchimpClient(apiKey);
```

Comment:
Argh, just realized what you meant in Slack. Preventing security bots from auto rotating the MC key, right? I get it now. Sorry, a little slow this weekend ;)

Instead of letting this leak into the actual src, can we remove all this and instead override Environment and hard code the key there? In `dr-hub`, check out `FunraiseToSfdcIT` for an example. In the constructor, we're hardcoding the key by overriding EnvironmentFactory -> Environment -> stripeClient. Here, it would look something like the following, right in `src/test/java/com/impactupgrade/nucleus/it/MailchimpCommunicationServiceIT.java`

```
  protected MailchimpCommunicationServiceIT() {
    super(new EnvironmentFactory("environment-it-sfdc-mailchimp") {
      @Override
      public Environment newEnv() {
        return new Environment() {
          @Override
          public MailchimpClient mailchimpClient(EnvironmentConfig.CommunicationPlatform mailchimpConfig) {
            mailchimpConfig.secretKey = [DECODE AND SET HERE];
            return new MailchimpClient(this, mailchimpConfig);
          }
        };
      }
    });
  }
```

Environment.mailchimpClient does not currently exist :( Entirely my fault -- I should have set it up that way from the beginning. Can you look at all uses of the `MailchimpClient` constructor and refactor them to go through a method in `Environment` instead? Something like:

`public MailchimpClient mailchimpClient(EnvironmentConfig.CommunicationPlatform mailchimpConfig) { return new MailchimpClient(this, mailchimpConfig); }`